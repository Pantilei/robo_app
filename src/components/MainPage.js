import React from "react";
import CardsList from "./CardsList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundry from "./ErrorBoundry";
import Header from "./Header";

class MainPage extends React.Component {
  componentDidMount() {
    this.props.fetchRobots();
    //console.log("componentDidMount");
  }
  filteredRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name.toLowerCase().includes(this.props.search.toLowerCase());
    });
  };
  render() {
    //console.log(this.props);

    //console.log("render");
    if (this.props.robots.length === 0) {
      return <h1>Loading...</h1>;
    }
    return (
      <div className="tc">
        <Header color={"red"} />
        <SearchBox
          onSearchChange={this.props.onSearch}
          inputValue={this.props.search}
        />
        <Scroll>
          <ErrorBoundry>
            <CardsList robots={this.filteredRobots()} />
          </ErrorBoundry>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
