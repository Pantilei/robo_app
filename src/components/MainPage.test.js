import React from "react";
import { shallow } from "enzyme";
import MainPage from "./MainPage";

let wrapper;

beforeEach(() => {
  const mockProps = {
    fetchRobots: jest.fn(),
    onSearch: jest.fn(),
    robots: [],
    search: ""
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

describe("MainComponent", () => {
  it("Renders MainPage without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Filters robots correctly", () => {
    const mockProps2 = {
      fetchRobots: jest.fn(),
      onSearch: jest.fn(),
      robots: [
        {
          id: 3,
          name: "John Doe",
          email: "john@gmail.com"
        }
      ],
      search: "john"
    };
    const wrapper2 = shallow(<MainPage {...mockProps2} />);

    expect(wrapper2.instance().filteredRobots()).toEqual([
      {
        id: 3,
        name: "John Doe",
        email: "john@gmail.com"
      }
    ]);
  });
});
