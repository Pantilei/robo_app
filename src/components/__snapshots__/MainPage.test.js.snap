// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent Renders MainPage without crashing 1`] = `ShallowWrapper {}`;

exports[`Renders MainPage without crashing 1`] = `ShallowWrapper {}`;
